action_server/go-wrapper/main.go

```diff
@@ -3,16 +3,82 @@ package main
 import (
 	"embed"
 	"fmt"
+	"io"
+	"net/http"
+	"net/url"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"runtime"
 	"strings"
+
+	"golang.org/x/mod/semver"
 )
 
 //go:embed all:assets/*
 var content embed.FS
 
+// Constants
+const ACTION_SERVER_LATEST_BASE_URL = "https://downloads.robocorp.com/action-server/releases/latest/"
+const VERSION_LATEST_URL = ACTION_SERVER_LATEST_BASE_URL + "version.txt"
+
+func getLatestVersion() (string, error) {
+	// Get the data from the URL
+	versionResponse, err := http.Get(VERSION_LATEST_URL)
+	if err != nil {
+		return "", err
+	}
+	defer versionResponse.Body.Close()
+
+	// Read the body content
+	versionInBytes, err := io.ReadAll(versionResponse.Body)
+	if err != nil {
+		return "", err
+	}
+	// Convert the byte slice to string
+	versionAsString := string(versionInBytes)
+	return strings.TrimSpace(versionAsString), nil
+}
+
+func checkAvailableUpdate(version string) {
+	// Get the latest version
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		fmt.Println("Verifying latest version failed:", err)
+		return
+	}
+	// Compare the given version with the latest one
+	compareResult := semver.Compare(strings.TrimSpace("v"+version), strings.TrimSpace("v"+latestVersion))
+
+	switch compareResult {
+	case -1:
```

Comment:
is this needed to be a switch if the only checked case is -1? Also this seems to be a case where early returns would look better as in:

```go
if compareResult != -1 {
    return
}
var actionOS, actionExe string
.
.
.
```

Reply:
Good point. Something that was kept for the other compare results.
Changed it.